csaf/src/model/store.rs

```diff
@@ -4,7 +4,15 @@ use std::path::{Path, PathBuf};
 
 /// create a distribution base directory
 pub fn distribution_base(base: impl AsRef<Path>, distribution: &Distribution) -> PathBuf {
-    base.as_ref().join(
-        utf8_percent_encode(distribution.directory_url.as_str(), NON_ALPHANUMERIC).to_string(),
-    )
+    if let Some(_directory_url) = &distribution.directory_url {
+        base.as_ref().join(
+            utf8_percent_encode(
+                distribution.directory_url.clone().unwrap().as_str(),
+                NON_ALPHANUMERIC,
+            )
+            .to_string(),
+        )
+    } else {
+        base.as_ref().to_path_buf()
```

Comment:
Wouldn't that write all documents into the same directory? If feels like we need a strategy for this. Also this would need to be aligned with the file source, which would read from such a store.